hikari/internal/enums.py

```diff
@@ -35,14 +35,45 @@
 _MAX_CACHED_MEMBERS: typing.Final[int] = 1 << 12
 
 
+class _DeprecatedAlias(typing.Generic[_T]):
+    __slots__ = ("_name", "_alias", "_removal_version")
+
+    def __init__(self, name: str, other: _T, removal_version: str) -> None:
+        self._name = name
+        self._alias = other
+        self._removal_version = removal_version
+
+    def __get__(self, instance: typing.Optional[_T], owner_class: _T) -> _T:
+        # Import kept in-line because of circular import issues
+        from hikari.internal import deprecation
+
+        deprecation.warn_deprecated(
+            self._name,
+            removal_version=self._removal_version,
+            additional_info=f"Use '{self._alias}' instead.",
+        )
+
+        return self._alias
+
+
+class deprecated:
+    """Used to denote that an enum member is a deprecated alias of another."""
+
+    __slots__ = ("alias", "removal_version")
+
+    def __init__(self, value: typing.Any, /, *, removal_version: str) -> None:
```

Comment:
this naming vs how it actually works seems a bit off here: you called it "value" but then pass the enum member's name instead.

Reply:
How this should be used is the following:

```py
class SomeEnum(int, enums.Enum):
    VAL = 1
    DEPRECATED = enum.deprecated(VAL, removal_version="...")
```

Since enum members are not "built" until the end of the class being defined, it is the same as doing

```py
class SomeEnum(int, enums.Enum):
    VAL = 1
    DEPRECATED = enum.deprecated(1, removal_version="...")
``` 

So, in this case the name is correct. In the case this still causes problems, would another name be more helpful? If so, which one? (I have no suggestions)